docs/java/how-to-set-activityoptions-in-java.md

```diff
@@ -0,0 +1,317 @@
+
+```
+
+Note that these Activity options will apply for all Activities within the Workflow.
+
+## Setting Ativity options within a Worker
+
+To set different options per Activity type, use `.setActivityOptions` with `WorkflowImplementationOptions`.
+Note that if you define options per Activity Type with `WorkflowImplementationOptions.setActivityOptions()`, setting them again specifically within `ActivityOptions` in a Workflow will override this setting.
+
+The following example shows how to set Activity options for Activity Types with `WorkflowImplementationOptions`.
+
+```java
+ WorkflowImplementationOptions options =
+                WorkflowImplementationOptions.newBuilder()
+                        // setActivityOptions allows you to set different ActivityOption per Activity type.
+                        // By default Activity type is the name of Activity method (with first letter capitalized.)
+                        .setActivityOptions(
+                                ImmutableMap.of(
+                                        "GetCustomerGreeting",
+                                        ActivityOptions.newBuilder()
+                                                // Set Activity execution timeout (including retries)
+                                                .setScheduleToCloseTimeout(Duration.ofSeconds(5))
+                                                .build(),
+                                        "EmailCustomerGreeting",
+                                        ActivityOptions.newBuilder()
+                                                // Set Activity execution timeout (single run)
+                                                .setStartToCloseTimeout(Duration.ofSeconds(2))
+                                                .setRetryOptions(
+                                                        RetryOptions.newBuilder()
+                                                                // ActivityTypeB activity type shouldn't retry on NPE
+                                                                .setDoNotRetry(NullPointerException.class.getName())
+                                                                .build())
+                                                .build()))
+                        .build();
+```
+
+## Activity Options
+
+The following table lists all `ActivityOptions` that can be configured for an Activity invocation.
+
+| Option                                                 | Required                                           | Type                     |
+| ------------------------------------------------------ | -------------------------------------------------- | ------------------------ |
+| [`setScheduleToCloseTimeout`](#scheduletoclosetimeout) | Yes (if `StartToCloseTimeout` is not specified)    | Duration                 |
+| [`setScheduleToStartTimeout`](#scheduletostarttimeout) | No                                                 | Duration                 |
```

Comment:
Would be nice to add somewhere that this timeout is non-retryable

Reply:
Updated